File: src/Budget/Budget.Core/Models/Records/CreateRecordModel.cs
using Budget.Core.Entities;

namespace Budget.Core.Models.Records
{
    public class CreateRecordModel
    {
        public string Note { get; set; }

        public decimal Amount { get; set; }

        public int AccountId { get; set; }

        public int CategoryId { get; set; }

        public int PaymentTypeId { get; set; }

        public RecordType RecordType { get; set; }

    }
}

File: web/src/app/shared/models/records/create-record.model.ts
export class CreateRecordModel {
  constructor(
    public note: string,
    public amount: number,
    public accountId: string,
    public categoryId: string,
    public paymentTypeId: string,
    public recordType: string,
    public recordDate: Date,
    public fromAccountId?: string,
  ) {}
}
